website/server/libs/groupTasks.js

```diff
@@ -0,0 +1,46 @@
+import * as Tasks from '../models/task';
+
+async function handleSharedCompletion (groupMemberTask) {
+  let masterTask = await Tasks.Task.findOne({
+    _id: groupMemberTask.group.taskId,
+  }).exec();
+
+  if (!masterTask || !masterTask.group || masterTask.type !== 'todo') return undefined;
+
+  if (masterTask.group.sharedCompletion === 'singleCompletion') {
```

Comment:
Let's break the code in both of these if branches into functions. 

`if (masterTask.group.sharedCompletion === 'singleCompletion') {` everything under this should be a function

`} else if (masterTask.group.sharedCompletion === 'allAssignedCompletion') {` everything under this should be in a function.